libtalek/subscription.go

```diff
@@ -37,22 +38,36 @@ func NewSubscription() (s *Subscription, err error) {
 	return
 }
 
+// nextBuckets returns the pair of buckets that will be used in the next poll or publish of this
+// topic given the current sequence number of the subscription.
+// The buckets returned by this method must still be wrapped by the NumBuckets config paramter of talek instance it is requested against.
+func (s *Subscription) nextBuckets() (uint64, uint64) {
+	seqNoBytes := make([]byte, 12)
+	_ = binary.PutUvarint(seqNoBytes, s.Seqno)
+
+	k0, k1 := s.Seed1.KeyUint128()
+	b1 := siphash.Hash(k0, k1, seqNoBytes)
+	k0, k1 = s.Seed2.KeyUint128()
+	b2 := siphash.Hash(k0, k1, seqNoBytes)
+
+	return b1, b2
+}
+
 func (s *Subscription) generatePoll(config *ClientConfig, seqNo uint64) (*common.ReadArgs, *common.ReadArgs, error) {
 	if s.SharedSecret == nil || s.SigningPublicKey == nil {
 		return nil, nil, errors.New("Subscription not fully initialized")
 	}
 
 	args := make([]*common.ReadArgs, 2)
-	seqNoBytes := make([]byte, 12)
-	_ = binary.PutUvarint(seqNoBytes, seqNo)
+	bucket1, bucket2 := s.nextBuckets()
+	bucket1 %= config.CommonConfig.NumBuckets
```

Comment:
I see your point. I didn't think of that.
I think you are right, adding a parameter of numBuckets is probably sufficient.
I don't think it's necessary to pass all of config right?